File: Stack/Questions/1614. Maximum Nesting Depth of the Parentheses.java
class Solution {
    public int maxDepth(String s) {
        // Get the length of the input string
        int sLength = s.length();

        // Initialize variables to track the maximum depth and the count of left parentheses
        int maxDepthOfParentheses = 0;
        int countLeftParentheses = 0;

        // Iterate through each character in the input string
        for (int i = 0; i < sLength; i++) {
            char currentCharacter = s.charAt(i);

            // If the current character is '(', increment the count of left parentheses
            if (currentCharacter == '(') {
                countLeftParentheses++;

                // Update the maximum depth based on the current count of left parentheses
                maxDepthOfParentheses = Math.max(maxDepthOfParentheses, countLeftParentheses);
            }
            // If the current character is ')', decrement the count of left parentheses
            else if (currentCharacter == ')') {
                countLeftParentheses--;
            }
        }

        // Return the maximum depth of parentheses
        return maxDepthOfParentheses;
    }
}

File: Stack/Parentheses/1614. Maximum Nesting Depth of the Parentheses.kt

[Apporoach (Count Method)]
------------------------------------------------------------------------------------------
import kotlin.math.max
class Solution {
    fun maxDepth(s: String): Int {
        var maximumNestingDepth:Int =0;
        var countDepth:Int = 0;

        for(character in s){
                if(character=='('){
                    countDepth++;
                }

                if(character==')'){
                    countDepth--;
                }

           maximumNestingDepth = max(countDepth,maximumNestingDepth);
        }


        return  maximumNestingDepth;
    }
}



[Apporoach (Stack Method)]
-----------------------------------------------------------------

import kotlin.math.max
class Solution {
    fun maxDepth(s: String): Int {

        val mainStack:Stack<Char> = Stack<Char>();

        var maximumNestingDepth:Int =0;

        s.forEach { character->
            if(character=='('){
                mainStack.push(character);
            }

            if(character==')'){
                mainStack.pop();
            }

            maximumNestingDepth = max(maximumNestingDepth,mainStack.size)



        }

        return maximumNestingDepth;


    }
}
